buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt

```diff
@@ -153,10 +153,10 @@ class CapabilitySpec {
     private
     fun ConfigurationContainer.forceUpgrade(to: String, version: String) = all {
         resolutionStrategy.dependencySubstitution {
-            providedBy.forEach { source ->
-                substitute(module(source))
-                    .because("Forceful upgrade of capability $name")
-                    .with(module("$to:$version"))
+            all {
+                if (providedBy.contains(requested.toString())) {
```

Comment:
The test is a bit weak compared to the original version, but I guess it's fine...